// jodit-angular/esm2015/lib/Events.js
import { Output, EventEmitter, Directive } from '@angular/core';
// tslint:disable-next-line:directive-class-suffix
import * as ɵngcc0 from '@angular/core';
export class Events {
    constructor() {
        // tslint:disable:no-output-on-prefix
        this.onChange = new EventEmitter();
        this.onBeforeEnter = new EventEmitter(false);
        this.onKeydown = new EventEmitter(false);
        this.onMousedown = new EventEmitter(false);
        this.onClick = new EventEmitter(false);
        this.onFocus = new EventEmitter();
        this.onBlur = new EventEmitter();
        this.onPaste = new EventEmitter(false);
        this.onResize = new EventEmitter();
        this.onBeforeCommand = new EventEmitter(false);
        this.onAfterCommand = new EventEmitter();
        this.onAfterExec = new EventEmitter();
        this.onAfterPaste = new EventEmitter();
        this.onChangeSelection = new EventEmitter();
    }
}
Events.ɵfac = function Events_Factory(t) { return new (t || Events)(); };
Events.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: Events, outputs: { onChange: "onChange", onBeforeEnter: "onBeforeEnter", onKeydown: "onKeydown", onMousedown: "onMousedown", onClick: "onClick", onFocus: "onFocus", onBlur: "onBlur", onPaste: "onPaste", onResize: "onResize", onBeforeCommand: "onBeforeCommand", onAfterCommand: "onAfterCommand", onAfterExec: "onAfterExec", onAfterPaste: "onAfterPaste", onChangeSelection: "onChangeSelection" } });
Events.propDecorators = {
    onChange: [{ type: Output }],
    onBeforeEnter: [{ type: Output }],
    onKeydown: [{ type: Output }],
    onMousedown: [{ type: Output }],
    onClick: [{ type: Output }],
    onFocus: [{ type: Output }],
    onBlur: [{ type: Output }],
    onPaste: [{ type: Output }],
    onResize: [{ type: Output }],
    onBeforeCommand: [{ type: Output }],
    onAfterCommand: [{ type: Output }],
    onAfterExec: [{ type: Output }],
    onAfterPaste: [{ type: Output }],
    onChangeSelection: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Events, [{
        type: Directive
    }], function () { return []; }, { onChange: [{
            type: Output
        }], onBeforeEnter: [{
            type: Output
        }], onKeydown: [{
            type: Output
        }], onMousedown: [{
            type: Output
        }], onClick: [{
            type: Output
        }], onFocus: [{
            type: Output
        }], onBlur: [{
            type: Output
        }], onPaste: [{
            type: Output
        }], onResize: [{
            type: Output
        }], onBeforeCommand: [{
            type: Output
        }], onAfterCommand: [{
            type: Output
        }], onAfterExec: [{
            type: Output
        }], onAfterPaste: [{
            type: Output
        }], onChangeSelection: [{
            type: Output
        }] }); })();
export const validEvents = [
    'onChange',
    'onBeforeEnter',
    'onKeydown',
    'onMousedown',
    'onClick',
    'onFocus',
    'onBlur',
    'onPaste',
    'onResize',
    'onBeforeCommand',
    'onAfterCommand',
    'onAfterExec',
    'onAfterPaste',
    'onChangeSelection',
];
