// jodit-angular/esm2015/lib/jodit-angular.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { JoditAngularComponent } from './jodit-angular.component';
import * as ɵngcc0 from '@angular/core';
export class JoditAngularModule {
}
JoditAngularModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: JoditAngularModule });
JoditAngularModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function JoditAngularModule_Factory(t) { return new (t || JoditAngularModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(JoditAngularModule, { declarations: function () { return [JoditAngularComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [JoditAngularComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(JoditAngularModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [JoditAngularComponent],
                exports: [
                    JoditAngularComponent
                ]
            }]
    }], null, null); })();
